recipes/hexl/all/conanfile.py

```diff
@@ -0,0 +1,129 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+import os
+
+required_conan_version = ">=1.43.0"
+
+class HexlConan(ConanFile):
+    name = "hexl"
+    license = "Apache-2.0"
+    homepage = "https://github.com/intel/hexl"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Intel Homomorphic Encryption (HE) Acceleration Library"
+    topics = ("homomorphic", "encryption")
+    settings = "os", "compiler", "build_type", "arch"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "experimental": [True, False],
+        "fpga_compatibility_dyadic_multiply": [True, False],
+        "fpga_compatibility_keyswitch":  [True, False]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "experimental": False,
+        "fpga_compatibility_dyadic_multiply": False,
+        "fpga_compatibility_keyswitch": False
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("hexl-{0}".format(self.version), self._source_subfolder)
```

Comment:
```suggestion
        get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```
from conan.tools.files import get
https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-layout-method

Reply:
Done